### PR TITLE
[WIP] Add Kohn-Sham Strictly Correlated Electrons method

### PR DESCRIPTION
**Pull request by DerkKooi**
Adds the Kohn-Sham Strictly Correlated Electrons method as an alternative. See refs. [1-5] and the notebook `example_kssce.ipynb` for details. The method works well for strongly correlated systems.

Some issues/points:
1. For cases in which the gap closes (e.g. $N=2$ singlet, where the density separates into two peaks) the convergence fails. In our work [5] we could achieve convergence using DIIS + dampening + level shifting. Do note that this happens for these systems with any functional that correctly describes them.
2. There are some choices in the method for using different interpolation methods, which are being pushed around using a dictionary `method_params`, but this is not very elegant. 
3. Not entirely clear to me yet which interpolation methods are best.

## References
1. F. Malet and P. Gori-Giorgi, Strong correlation in Kohn-Sham density functional theory, Phys. Rev. Lett. 109, 246402 (2012). DOI: [10.1103/PhysRevLett.109.246402](https://dx.doi.org/10.1103/PhysRevLett.109.246402)
4. F. Malet, A. Mirtschink, J. C. Cremon, S. M. Reimann, and P. Gori-Giorgi. Kohn-Sham density functional theory for quantum wires in arbitrary correlation regimes. Phys. Rev. B 87, 115146 (2013) DOI: [10.1103/PhysRevB.87.115146](http://dx.doi.org/10.1103/PhysRevB.87.115146)
5. A. Mirtschink, M. Seidl, and P. Gori-Giorgi. Derivative discontinuity in the strong-interaction limit of density functional theory. Phys. Rev. Lett.  111, 126402 (2013) DOI: [10.1103/PhysRevLett.111.126402](http://dx.doi.org/10.1103/PhysRevLett.111.126402)
6. F. Malet, A. Mirtschink, K. J. H. Giesbertz, L. O. Wagner, and P. Gori-Giorgi. Exchange-correlation functionals from the strongly-interacting limit of DFT: Applications to model chemical systems. Phys. Chem. Chem. Phys. 16, 14551 (2014) DOI: [10.1039/c4cp00407h](http://dx.doi.org/10.1039/c4cp00407h)
7. A. Marie, D. P. Kooi, J. Grossi, M. Seidl, Z. Musslimani, K.J.H. Giesbertz, P. Gori-Giorgi. Real space Mott-Anderson electron localization with long-range interactions. Physical Review Research 4, 043192 (2022) DOI: [10.1103/PhysRevResearch.4.043192](https://dx.doi.org/10.1103/PhysRevResearch.4.043192)

### CODE REVIEW

**Review by DrMattHodgson — APPROVED** [review]
This pull request introduces the 'stricktly correlated electrons' (SCE) approach as an additional approximation to iDEA's existing methods. The key addition is the file 'kssce.py', which contains code implementing various functions for the computation of the Kohn-Sham SCE method. These functions enable interpolation of the charge density and cumulant, computation of co-motion functions, calculation of the SCE potential energy operator, SCE energy, Hamiltonian, and total energy. The code employs interpolation techniques and numerical integration to approximate the required quantities accurately.

File: iDEA/methods/kssce.py [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+        n_interp = interpolate_n(s.x, n, interp=method_params["interp_n"])
+        # Compute the cumulant of the charge density
+        Ne = n_interp.antiderivative()(s.x)
+
+        # Compute co-motion functions
+        f = compute_comotion_functions(
+            n, Ne, s.x, s.count, method_params["interp_invNe"]
+        )
+
+        # Compute the derivative of the SCE potential
```
Comment (JackWetherell) [inline iDEA/methods/kssce.py:180]: Is something missing here?

**Review by JackWetherell — COMMENTED** [review]
This is great! Can you confirm there is no missing code below the comment marked? Then I will go ahead and merge.

File: iDEA/methods/kssce.py (hunk 1 above)
Comment (DerkKooi) [reply iDEA/methods/kssce.py:180]: No, the whole part after it is the computation of the SCE potential. See lines 202 and on that yield v_sce.

**Comment by JackWetherell** [comment]
Thank you @DerkKooi !!

**Pull request merged**